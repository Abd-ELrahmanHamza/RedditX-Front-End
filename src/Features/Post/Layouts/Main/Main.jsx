// Import styled components
import { Title, Container } from "./Main.styled";

// Import components and layouts
import ChooseCommunity from "Features/Post/Components/ChooseCommunity/ChooseCommunity";
import Tabs from "Features/Post/Layouts/Tabs/Tabs";

// Import hooks
import { useState } from "react";

// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/selectedDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";

/**
 * Main component (the main section in the create post page)
 *
 * @returns {React.Component} Main component (the main section in the create post page)
 */
const Main = () => {
  // State to store the selected community
  const [selectedDestination, setSelectedDestination] = useState(null);
  return (
    <SubmitDestinationProvider>
      <CreatePostTitleProvider>
        <Container>
          <Title>
            <h4>Create a post</h4>
          </Title>
          <ChooseCommunity
            selectedDestination={selectedDestination}
            setSelectedDestination={setSelectedDestination}
          />
          <Tabs selectedDestination={selectedDestination} />
        </Container>
      </CreatePostTitleProvider>
    </SubmitDestinationProvider>
  );
};

export default Main;
