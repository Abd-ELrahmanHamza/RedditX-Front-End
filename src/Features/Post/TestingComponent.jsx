// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/submitDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";
import { CreatePostFlagsProvider } from "Features/Post/Contexts/createPostFlags";
import { CreatePostAttachmentsProvider } from "Features/Post/Contexts/createPostAttachments";
import { CreatePostFlairsProvider } from "Features/Post/Contexts/createPostFlairs";
import { CreatePostTextProvider } from "Features/Post/Contexts/createPostText";
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  const auth = useAuth();
  useEffect(() => {
    auth.login({
      expiresIn: 86400,
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbFR5cGUiOiJiYXJlIGVtYWlsIiwidXNlcm5hbWUiOiJ0Ml9ob21lIiwiaWF0IjoxNjY5OTI0MDEzLCJleHAiOjE2NzAzNTYwMTN9.FDa59vJi5gWy4dQVkEtgyCfm3Pp2BtHkPz0_AfNVKgY",
      username: "t2_home",
    });
  }, []);

  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SubmitDestinationProvider>
          <CreatePostTitleProvider>
            <CreatePostFlagsProvider>
              <CreatePostAttachmentsProvider>
                <CreatePostFlairsProvider>
                  <CreatePostTextProvider>{children}</CreatePostTextProvider>
                </CreatePostFlairsProvider>
              </CreatePostAttachmentsProvider>
            </CreatePostFlagsProvider>
          </CreatePostTitleProvider>
        </SubmitDestinationProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
