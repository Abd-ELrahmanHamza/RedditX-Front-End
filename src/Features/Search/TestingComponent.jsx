// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { SafeContextProvider } from "./Contexts/SafeSearchContext/Safe-context";
import { SearchContextProvider } from "./Contexts/SearchWordContext/Search-context";

const TestingComponent = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SearchContextProvider>
          <SafeContextProvider>{children}</SafeContextProvider>
        </SearchContextProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default TestingComponent;
