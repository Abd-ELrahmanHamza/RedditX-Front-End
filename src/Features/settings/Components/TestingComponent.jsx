import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
