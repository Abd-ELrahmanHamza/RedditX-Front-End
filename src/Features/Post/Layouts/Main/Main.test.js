import { render, screen } from "@testing-library/react";
import TestingComponent from "Features/Post/TestingComponent";

// Import components
import Main from "./Main";

jest.mock("Features/Post/Services/submitPost", () => (fetchData, auth) => {});

describe("Main section", () => {
  it("renders Main component", () => {
    render(
      <TestingComponent>
        <Main />
      </TestingComponent>
    );
    const headingElement = screen.getByText("Create a post");
    expect(headingElement).toBeVisible();
  });
});
