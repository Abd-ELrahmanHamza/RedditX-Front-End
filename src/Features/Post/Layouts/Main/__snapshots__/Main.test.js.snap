// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main section should render without crashing 1`] = `
<SubmitDestinationProvider>
  <CreatePostTitleProvider>
    <styled.div>
      <styled.div>
        <h4>
          Create a post
        </h4>
      </styled.div>
      <ChooseCommunity
        selectedDestination={null}
        setSelectedDestination={[Function]}
      />
      <Tabs
        selectedDestination={null}
      />
    </styled.div>
  </CreatePostTitleProvider>
</SubmitDestinationProvider>
`;
